lib/jsdom/living/nodes/HTMLFrameElement-impl.js

```diff
@@ -54,7 +57,26 @@ function fetchFrame(serializedURL, frame, document, contentDoc) {
     contentDoc._encoding = encoding;
 
     const html = whatwgEncoding.decode(data, contentDoc._encoding);
-    contentDoc._htmlToDom.appendToDocument(html, contentDoc);
+
+    try {
+      parseIntoDocument(html, contentDoc);
+    } catch (error) {
+      if (
+        error instanceof DOMException &&
+        error.code === DOMException.SYNTAX_ERR &&
+        contentDoc._parsingMode === "xml"
+      ) {
+        // As defined (https://html.spec.whatwg.org/#read-xml) parsing error in XML document may be reported inline by
+        // mutating the document.
+        const element = document.createElementNS("http://www.mozilla.org/newlayout/xml/parsererror.xml", "parsererror");
+        element.textContent = error.message;
+
+        document.body.appendChild(element);
```

Comment:
The spec does not provide much detail regarding this, but browsers seem to differ in their handling of parsing errors. Chrome and Safari seem to append an error to the existing document, while Firefox appears to throw it away and return a new document with only a `<parsererror>` element ([example](http://malform.no/testing/html5/bom/ascii.us-ascii.xhtml)). Perhaps we should follow Firefox here to simplify our error handling.

Reply:
I like that idea, let me try it.